Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,152 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - SnakeCase for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating an on-disk movie database :file:`tutorial.db`
+about Monty Python movies.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, create a new database to hold the movie data,
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to the database :file:`tutorial.db`,
+implicitly creating it in the current working directory if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
+Now, create a database table ``movie`` with columns for title,
+release year, and review score.
+The `flexible typing`_ feature of SQLite makes typenames optional;
+for simplicity, just use column names in the table declaration.
+Execute the SQL statement
+by calling :meth:`con.execute() <Cursor.execute>`::
 
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+   cur.execute("CREATE TABLE Movie(Title, Year, Score)")
 
-   # Save (commit) the changes
-   con.commit()
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite does not recognise that variant.
```

Comment:
Are the old versions common enough?  Is `sqlite_master` deprecated?
Maybe we could use `sqlite_schema` directly, or add a footnote or parenthetical to mention `sqlite_master`.

Reply:
`sqlite_master` is deprecated (in docs only). It will never be removed from SQLite 3.x.y; they have a super strict backwards compatibility policy. It is an alias to `sqlite3_schema`. I'm unsure when they changed it. Last time I checked, the changelog did not mention it. It may be worth it to mention both names, though.

Reply:
> I'm unsure when they changed it. Last time I checked, the changelog did not mention it.

<details>
<summary>Found out when they changed it</summary>

```
$ for N in $(seq 8 39); do echo -n "Checking SQLite 3.$N.0... " && (~/install/sqlite-3.$N.0/bin/sqlite3 :memory: "select * from sqlite_schema" &>/dev/null && echo ok || echo fail); done
Checking SQLite 3.8.0... fail
Checking SQLite 3.9.0... fail
Checking SQLite 3.10.0... fail
Checking SQLite 3.11.0... fail
Checking SQLite 3.12.0... fail
Checking SQLite 3.13.0... fail
Checking SQLite 3.14.0... fail
Checking SQLite 3.15.0... fail
Checking SQLite 3.16.0... fail
Checking SQLite 3.17.0... fail
Checking SQLite 3.18.0... fail
Checking SQLite 3.19.0... fail
Checking SQLite 3.20.0... fail
Checking SQLite 3.21.0... fail
Checking SQLite 3.22.0... fail
Checking SQLite 3.23.0... fail
Checking SQLite 3.24.0... fail
Checking SQLite 3.25.0... fail
Checking SQLite 3.26.0... fail
Checking SQLite 3.27.0... fail
Checking SQLite 3.28.0... fail
Checking SQLite 3.29.0... fail
Checking SQLite 3.30.0... fail
Checking SQLite 3.31.0... fail
Checking SQLite 3.32.0... fail
Checking SQLite 3.33.0... ok
Checking SQLite 3.34.0... ok
Checking SQLite 3.35.0... ok
Checking SQLite 3.36.0... ok
Checking SQLite 3.37.0... ok
Checking SQLite 3.38.0... ok
Checking SQLite 3.39.0... ok
```

</details>

Reply:
So, they added `sqlite_schema` in 3.33.0, which is fairly new. I don't think we should introduce the option of using `sqlite_schema`. IIRC, Diátaxis suggests to avoid options and alternatives in tutorials.

If we add a _SQLite tips and tricks_ howto, we can mention it there.

Reply:
Or in the howto I proposed above about listing tables and columns.

Reply:
Agreed. See also my latest post on this PR.